src/nvim/ex_cmds.c

```diff
@@ -3533,7 +3533,7 @@ void do_sub(exarg_T *eap)
 
         // 3. Substitute the string. Don't do this while incsubstitution and
         //    there's no word to replace by eg : ":%s/pattern"
-        if (!(eap->is_live && sub[0] == '\0' && !last_is_slash)) {
+        if (!eap->is_live || sub[0] != '\0' || last_is_slash) {
```

Comment:
Hmm, `last_is_slash` is wrong in any case, we should check for any delimiter, maybe like this
```diff
-  const bool last_is_slash = !(eap->arg == NUL
-                               || eap->arg[STRLEN(eap->arg) - 1] != '/');
+  bool has_second_delim = false;
 
   cmd = eap->arg;
   if (!global_busy) {
@@ -2989,6 +2988,7 @@ void do_sub(exarg_T *eap)
         which_pat = RE_SEARCH;              /* use last '/' pattern */
       pat = (char_u *)"";                   /* empty search pattern */
       delimiter = *cmd++;                   /* remember delimiter character */
+      has_second_delim = true;
     } else {          /* find the end of the regexp */
       if (p_altkeymap && curwin->w_p_rl)
         lrF_sub(cmd);
@@ -2996,8 +2996,10 @@ void do_sub(exarg_T *eap)
       delimiter = *cmd++;                   /* remember delimiter character */
       pat = cmd;                            /* remember start of search pat */
       cmd = skip_regexp(cmd, delimiter, p_magic, &eap->arg);
-      if (cmd[0] == delimiter)              /* end delimiter found */
+      if (cmd[0] == delimiter) {             /* end delimiter found */
         *cmd++ = NUL;                       /* replace it with a NUL */
+        has_second_delim = true;
+      }
     }
 
     /*
@@ -3536,7 +3538,7 @@ void do_sub(exarg_T *eap)
 
         // 3. Substitute the string. Don't do this while incsubstitution and
         //    there's no word to replace by eg : ":%s/pattern"
-        if (!(eap->is_live && sub[0] == '\0' && !last_is_slash)) {
+        if (!(eap->is_live && !has_second_delim)) {
           if (do_count) {
             // prevent accidentally changing the buffer by a function
             save_ma = curbuf->b_p_ma;
@@ -3874,7 +3876,7 @@ skip:
   if (!kl_empty(lmatch)) {
     // we did incsubstitute only if we had no word to replace by
     // by and no ending slash
-    if (!(eap->is_live && sub[0] == '\0' && !last_is_slash)) {
+    if (!(eap->is_live && !has_second_delim)) {
       sub_done = 1;
     }
     if (pat != NULL && *p_ics != NUL && eap->is_live) 
```